pkg/controller/logstorage/logstorage_controller.go

```diff
@@ -483,6 +500,65 @@ func (r *ReconcileLogStorage) Reconcile(request reconcile.Request) (reconcile.Re
 	return reconcile.Result{}, nil
 }
 
+// getAuthentication
+func (r *ReconcileLogStorage) getAuthentication(ctx context.Context) (interface{}, error) {
+	var authenticationConfig interface{}
+
+	authenticationCR := &operatorv1.Authentication{}
+	if err := r.client.Get(ctx, utils.DefaultTSEEInstanceKey, authenticationCR); err != nil {
+		if !errors.IsNotFound(err) {
+			log.Error(err, err.Error())
+			r.status.SetDegraded("An error occurred retrieving the Authentication", err.Error())
+			return nil, err
+		}
+	} else {
+		if authenticationCR.Spec.Method == operatorv1.AuthMethodOIDC {
+			if authenticationCR.Spec.OIDC == nil {
+				return nil, fmt.Errorf("authentication method is OIDC but no OIDC configuration was provided")
+			}
+
+			wellKnownConfig, err := oidc.LookupWellKnownConfig(authenticationCR.Spec.OIDC.IssuerURL)
+			if err != nil {
+				return nil, err
```

Comment:
Is it right to return malformed json errors? Wouldn't we want to repackage the error in a nicer message?